src/plugins/client.ts

```diff
@@ -0,0 +1,45 @@
+import fg from "fast-glob";
+import fs from "node:fs/promises";
+import type { QwikPWAContext } from "../context";
+import type { Plugin } from "vite";
+
+export default function ClientPlugin(ctx: QwikPWAContext): Plugin {
+  return {
+    name: "qwik-pwa:client",
+    enforce: "post",
+    writeBundle: {
+      sequential: true,
+      order: "post",
+      async handler(_, bundle) {
+        if (ctx.target !== "client") {
+          return;
+        }
+        const publicDirAssets = await fg.glob("**/*", { cwd: ctx.publicDir });
+        // the q-*.js files are going to be handled by qwik itself
+        const emittedAssets = Object.keys(bundle).filter(
+          (key) => !/.*q-.*\.js$/.test(key),
+        );
+
+        const routes = ctx.qwikCityPlugin.api.getRoutes();
+        const swCode = await fs.readFile(ctx.swClientDistPath, "utf-8");
+        const swCodeUpdate = `
+        const publicDirAssets = ${JSON.stringify(publicDirAssets)};
+        const emittedAssets = ${JSON.stringify(emittedAssets)};
```

Comment:
add here pwa assets

Reply:
Could you elaborate more on this?

Reply:
we're using workbox precaching: where are the generated pwa icons entries here? public and emitted assets being used in the sw recipe to add them to the manifest => offline will not work (I need to do some tests)